File: functions/ffmpeg/variants/go/2_vp5_adjust_4.go
package main

func vp5_adjust(v int32, t int32) int32 {
    s1 := v >> 31
    v = ((v ^ s1) - s1) * int32(v < 2*t)
    adjusted := v - t
    s2 := adjusted >> 31
    adjusted = (adjusted ^ s2) - s2
    return (t - adjusted + s1) ^ s1
}


func main(){
    vp5_adjust(0,0)
}

File: functions/ffmpeg/variants/c/2_vp5_adjust_4.c
#include <stdint.h>

static int vp5_adjust(int v, int t)
{
    int s1 = v >> 31;
    int abs_v = (v ^ s1) - s1;
    if (abs_v >= 2 * t) abs_v = 0;
    abs_v -= t;
    int s2 = abs_v >> 31;
    abs_v = (abs_v ^ s2) - s2;
    return ((t - abs_v) ^ s1) + s1;
}


int main(){
    vp5_adjust(0,0);
}
